http/http-api/src/test/groovy/com/okta/commons/http/HttpHeadersTest.groovy

```diff
@@ -0,0 +1,440 @@
+
+        headerMap = httpHeaders.toSingleValueMap()
+        assertEquals headerMap.size(), 2
+        assertEquals headerMap.get("Pragma"), "no-cache"
+        assertEquals headerMap.get("Content-Type"), MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    }
+
+    @Test
+    void testEmpty() {
+
+        assertTrue httpHeaders.isEmpty()
+
+        httpHeaders.setLocation(new URI("https://mylocation.com"))
+
+        assertFalse httpHeaders.isEmpty()
+
+        httpHeaders.clear()
+
+        assertTrue httpHeaders.isEmpty()
+    }
+
+    @Test
+    void testContainsKey() {
+
+        httpHeaders.setLocation(new URI("https://mylocation.com"))
+        assertTrue httpHeaders.containsKey("Location")
+    }
+
+    @Test
+    void testContainsValue() {
+
+        httpHeaders.setLocation(new URI("https://mylocation.com"))
+        assertTrue httpHeaders.containsValue(["https://mylocation.com"])
+    }
+
+    @Test
+    void testPut() {
+
+        httpHeaders.put("Pragma", ["no-cache"])
+        assertEquals httpHeaders.getPragma(), "no-cache"
+    }
+
+    @Test
+    void testPutAll() {
+
+        httpHeaders.putAll([
+            "Pragma":["no-cache"],
+            "ETag":["123456789"]
+        ])
+        assertEquals httpHeaders.getPragma(), "no-cache"
+        assertEquals httpHeaders.getETag(), "123456789"
+    }
+
+    @Test
+    void testKeySet() {
+
+        httpHeaders.putAll([
+            "Pragma":["no-cache"],
+            "ETag":["123456789"]
+        ])
+
+        def keySet = httpHeaders.keySet()
+        assertEquals keySet.size(), 2
+        assertTrue keySet.contains("Pragma")
+        assertTrue keySet.contains("ETag")
+    }
+
+    @Test
+    void testValues() {
+
+        httpHeaders.putAll([
+            "Pragma":["no-cache"],
+            "ETag":["123456789"]
+        ])
+
+        def values = httpHeaders.values()
+        assertEquals values.size(), 2
+        assertTrue values.contains(["no-cache"])
+        assertTrue values.contains(["123456789"])
+    }
+
+    @Test
+    void testToString() {
+
+        httpHeaders.putAll([
+            "Pragma":["no-cache"],
+            "ETag":["123456789"]
+        ])
+        assertEquals httpHeaders.toString(), "{Pragma=[no-cache], ETag=[123456789]}"
+    }
+
+    @Test
+    void testEquals() {
+
+        assertTrue httpHeaders.equals(httpHeaders)
+
+        def notHeaders = "NOT_HEADERS"
+        assertFalse httpHeaders.equals(notHeaders)
+
+        def other = new HttpHeaders()
+        other.setExpires(3600)
+        httpHeaders.setExpires(3600)
+        assertTrue other.equals(httpHeaders)
+    }
+
+    @Test
+    void testHashCode() {
+
+        def location = "https://mylocation.com"
+
+        def other = new HttpHeaders()
+        other.setExpires(3600)
+        other.setLocation(new URI(location))
+
+        httpHeaders.setExpires(3600)
+        httpHeaders.setLocation(new URI(location))
+
+        assertEquals httpHeaders.hashCode(), other.hashCode()
+    }
+
+//    @Test
```

Comment:
Can this be added back in or should it be deleted?

Reply:
Fixed, I was originally thinking about moving that method out of HttpHeaders, but added it back in and forgot about that commented test.  Good find!